draft-packages/rating-scale-legacy/KaizenDraft/RatingScaleLegacy/RatingScaleLegacy.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useState, createRef } from "react"
+
+import classNames from "classnames"
+import { Paragraph } from "@kaizen/component-library"
+import determineSelectionFromKeyPress from "./helpers/determineSelectionFromKeyPress"
+import { Scale, ScaleItem, ScaleValueResponse } from "./types"
+
+const styles = require("./styles.module.scss")
+
+type ItemRefs = Array<{
+  value: ScaleValueResponse
+  ref: { current: null | HTMLDivElement }
+}>
+
+export interface RatingScaleProps {
+  labelId: string
+  scale: Scale
+  selectedItem: ScaleItem | null
+  questionId: string
+  reversed?: boolean
+  validationError?: boolean
+  validationErrorId?: string
+  onSelect: (value: ScaleItem | null) => void
+}
+
+export const RatingScaleLegacy = ({
+  scale,
+  questionId,
+  selectedItem,
+  reversed,
+  onSelect,
+  validationError,
+  validationErrorId,
+  labelId,
+}: RatingScaleProps) => {
+  const [hoveredItem, setHoveredItem] = useState<ScaleItem | null>(null)
+  const itemRefs: ItemRefs = scale.map(s => ({
+    value: s.value,
+    ref: createRef<HTMLDivElement>(),
+  }))
+
+  const handleRadioClick = (item: ScaleItem): void => {
+    const newValue =
+      selectedItem && item.value === selectedItem.value
+        ? scale.find(s => s.value === -1) || null
+        : item
+    onSelect(newValue)
+
+    // Remove hover state
+    setHoveredItem(null)
+  }
+
+  const handleRadioMouseEnter = (item: ScaleItem): void => {
+    if (selectedItem && selectedItem.value > 0) {
+      return // Disable hover when there is a selection
+    }
+    setHoveredItem(item)
+  }
+
+  const handleRadioMouseLeave = (): void => {
+    if (selectedItem && selectedItem.value > 0) {
+      return // Disable hover when there is a selection
+    }
+    setHoveredItem(null)
+  }
+
+  /**
+   * Because the radios have been built with divs, we need to add the keyboard functionality manually
+   */
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLDivElement>,
+    focusedItem: ScaleItem
+  ): void => {
+    const newPosition = determineSelectionFromKeyPress(
+      event.keyCode,
+      selectedItem,
+      focusedItem
+    )
+    if (newPosition) {
+      event.preventDefault()
+
+      // XHR
+      onSelect(scale.find(s => s.value === newPosition) || null)
+
+      // Update focus
+      const itemRef = itemRefs.find(item => item.value === newPosition)
+      itemRef && itemRef.ref.current && itemRef.ref.current.focus()
+    }
+  }
+
+  let legend = "Not rated"
+  if (hoveredItem) {
+    legend = hoveredItem.label
+  } else if (selectedItem) {
+    legend = selectedItem.label
+  }
+
+  return (
+    <div
+      className={classNames(styles.container, {
+        [styles.rated]: selectedItem && selectedItem.value > 0,
+        [styles.reversed]: reversed,
+      })}
+      aria-labelledby={labelId}
+      role="radiogroup"
+      data-focus-to-on-navigate="true"
```

Comment:
This data attribute can be removed. It's specific to Capture. I'll have a find a way to get it in when it comes time to use this component in Capture, but that can be handled later.